exchanges/btcmarkets/btcmarkets_test.go

```diff
@@ -542,18 +543,51 @@ func TestWsTicker(t *testing.T) {
 	}
 }
 
-func TestWsTrade(t *testing.T) {
-	pressXToJSON := []byte(` { "marketId": "BTC-AUD",
-    "timestamp": "2019-04-08T20:54:27.632Z",
-    "tradeId": 3153171493,
-    "price": "7370.11",
-    "volume": "0.10901605",
-    "side": "Ask",
-    "messageType": "trade"
-  }`)
-	err := b.wsHandleData(pressXToJSON)
-	if err != nil {
-		t.Error(err)
+func TestWSTrade(t *testing.T) {
+	t.Parallel()
+
+	b := new(BTCMarkets) //nolint:govet // Intentional shadow to avoid future copy/paste mistakes
+	require.NoError(t, testexch.Setup(b), "Test instance Setup must not error")
+	fErrs := testexch.FixtureToDataHandlerWithErrors(t, "testdata/wsAllTrades.json", b.wsHandleData)
+	for _, fErr := range fErrs {
+		require.ErrorIs(t, fErr.Err, order.ErrSideIsInvalid)
+	}
+	close(b.Websocket.DataHandler)
```

Comment:
🚧 
* Need to check the length is correct
* Prefer to not use range because the causes are distinct, and also because if we add more error handling tests it won't scale
```suggestion
	fErrs := testexch.FixtureToDataHandlerWithErrors(t, "testdata/wsAllTrades.json", b.wsHandleData)
	require.Equal(t, 2, len(fErrs), "Must get correct number of errors from wsHandleData")	
	assert.ErrorIs(t, fErr[0].Err, order.ErrSideIsInvalid, "Side.UnmarshalJSON errors must propagate correctly")
	assert.ErrorContains(t, fErr[0].Err, "Pentagon", "Side.UnmarshalJSON errors must propagate correctly")
	assert.ErrorIs(t, fErr[0].Err, order.ErrSideIsInvalid, "wsHandleData errors must propagate correctly")
	assert.ErrorContains(t, fErr[0].Err, "Any", "wsHandleData errors must propagate correctly")
	close(b.Websocket.DataHandler)
```